RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -109,19 +177,78 @@ extension PurchaseHandler {
             }
 
             return result
+
         } catch {
             self.purchaseError = error
             throw error
         }
     }
 
+    @MainActor
+    func performExternalPurchaseLogic(package: Package) async throws -> PurchaseResultData {
+        Logger.debug(Strings.executing_external_purchase_logic)
+
+        guard let externalPurchaseMethod = self.performPurchase else {
+            throw PaywallError.performPurchaseAndRestoreHandlersNotDefined(missingBlocks: "performPurchase is")
+        }
+
+        self.packageBeingPurchased = package
+        self.purchaseResult = nil
+        self.purchaseError = nil
+
+        defer {
+            self.restoreInProgress = false
+            self.actionInProgress = false
+        }
+
+        self.startAction()
+
+        let result = await externalPurchaseMethod(package)
+
+        if result.userCancelled {
+            self.trackCancelledPurchase()
+        }
+
+        if let error = result.error {
+            self.purchaseError = error
+            throw error
+        }
+
+        let resultInfo: PurchaseResultData = (transaction: nil,
+                                             customerInfo: try await self.purchases.customerInfo(),
+                                            userCancelled: result.userCancelled)
```

Comment:
When we do an external purchase, is transaction `nil`? Is the customer info retrieved from purchases correct? Could use some experienced opinions here!

Reply:
from an external purchase: 
- customer info should be correct after we get the purchase in our backend, which will depend basically on our purchase observing code. So depending on when you're checking, maybe not. If we wanted to guarantee correctness, then... it's tricky. We _could_ check if the result is success, invalidate customerInfo cache, and refresh customerInfo, but that might create a redundant backend request if the customerInfo had already been refreshed (🐼). Otherwise we'd need some other hook to prevent that redundancy, nothing comes to mind for how to do it though. 
- transaction data... we could require it as part of the external purchase method result? WDYT? Although presumably if you already made the purchase yourself, you already have the transaction in the first place.

Reply:
Looking at this again, the only thing that uses `purchaseResult` property (which is declared as `fileprivate`) is our tests. They never look at `transaction:`, and for `customerInfo:`, they just check if it's the same as the mock that was passed in. So I think this is all fine as is.